awsiot/iotshadow.py

```diff
@@ -1327,16 +1327,14 @@ def __init__(self, *args, **kwargs):
     def from_payload(cls, payload):
         # type: (typing.Dict[str, typing.Any]) -> ShadowState
         new = cls()
-        val = payload.get('desired')
-        if val is not None:
-            new.desired = val
-        val = payload.get('reported')
-        if val is not None:
-            new.reported = val
-        if new.desired == None:
-            new.desired_is_nullable = True
-        if new.reported == None:
-            new.reported_is_nullable = True
+        if 'desired' in payload:
+            new.desired = payload['desired']
+            new.desired_is_nullable = new.desired is None
```

Comment:
add tests to confirm that this nullable stuff works like we expect

Reply:
Added test that checks that the to and from are resulting in the same data. Failed without implemented changes, succeeds with changes.